### PR TITLE
feat/BaseProductV2: create new product base type, efficient product factory

### PR DESCRIPTION
**Pull request by leonardishere**

### CODE REVIEW

File: contracts/products/BaseProductV2.sol [hunk 1]
```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+import "../GovernableInitializable.sol";
+import "../interface/IPolicyManager.sol";
+import "../interface/IRiskManager.sol";
+import "../interface/ITreasury.sol";
```
Comment (ssozuer) [inline contracts/products/BaseProductV2.sol:10]: Idk, Is it ok to use non-upgradeable contract imports in an upgradeable contract?

File: contracts/PolicyDescriptor.sol [hunk 2]
```diff
@@ -19,8 +19,9 @@ contract PolicyDescriptor is IPolicyDescriptor {
      * @return description The URI of the ERC721-compliant metadata.
      */
     function tokenURI(IPolicyManager policyManager, uint256 policyID) external view override returns (string memory description) {
-        address product = policyManager.getPolicyProduct(policyID);
-        string memory productName = IProduct(product).name();
+        //address product = policyManager.getPolicyProduct(policyID);
```
Comment (ssozuer) [inline contracts/PolicyDescriptor.sol:22]: If we remove the "product name" how do we distinguish products?

File: contracts/products/BaseProductV2.sol (hunk 1 above)
Comment (leonardishere) [reply contracts/products/BaseProductV2.sol:10]: its a proxy factory not an upgradeable contract, but either way it depends on your usage. thats why I had to make GovernableInitializable

File: contracts/PolicyDescriptor.sol (hunk 2 above)
Comment (leonardishere) [reply contracts/PolicyDescriptor.sol:22]: on chain it doesnt matter. off chain we'll use some sort of mapping

**Pull request merged**